### PR TITLE
DAOS-2153 debug: Update to gurt debug log version 2

### PR DESCRIPTION
**Pull request by jolivier23**
This is an optimization when logging is disabled as it
bypasses the bitwise arithmetic and several checks,
replacing with a single check for all but the first
check.
Also, use daos_ prefix for log variables
Updates to latest cart to get the feature
Updates scons_local to latest to use github.com

Change-Id: I8940de203e0911a2ab097a6566a6624514417a67
Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: src/common/debug.c [hunk 1]
```diff
@@ -35,103 +35,24 @@
 #define DAOS_DBG_MAX_LEN	(32)
 #define DAOS_FAC_MAX_LEN	(128)
 
+
```
Comment (liuxuezhao) [inline src/common/debug.c]: just an extra blank line?

**Comment by gnailzenh** [comment]
It failed functional test?

**Comment by jolivier23** [comment]
I previously updated to earlier version of cart and it passed.  Then I updated to latest cart and hit the bug Alex mentioned in chat.   He has a patch for that.    I think I will update to that version.

But in this case, I've reverted cart version and think I'm just hitting the intermittent failure with pool create

**Comment by brianjmurrell** [comment]
> But in this case, I've reverted cart version and think I'm just hitting the intermittent failure with pool create

The intermittent failure has been isolated to a single one of the test clusters and it has since been removed from the pool.  It would be worthwhile using "Restart from Stage" to rerun the Functional tests on this if there is any doubt.

**Pull request merged**